ReactiveCocoa/CocoaAction.swift

```diff
@@ -37,9 +35,11 @@ public final class CocoaAction<Sender>: NSObject {
 			producer.start()
 		}
 
-		isEnabled = action.isEnabled
-		isExecuting = action.isExecuting
-		
+		isEnabled = Property(initial: action.isEnabled.value,
+		                     then: action.isEnabled.producer.observe(on: UIScheduler()))
```

Comment:
Do you think it would make sense for `PropertyProtocol` to have a version of `observe(on:)`? I've been wondering if there's a potential race between reading `property.value` and starting `property.producer`.